gateway-ha/src/main/java/io/trino/gateway/ha/resource/LoginResource.java

```diff
@@ -107,7 +108,16 @@ public Response processRESTLogin()
     public Response restUserinfo(@Context SecurityContext securityContext)
     {
         LbPrincipal principal = (LbPrincipal) securityContext.getUserPrincipal();
-        List<String> roles = List.of(principal.getMemberOf().map(String::toLowerCase).orElse("").split("_"));
+        List<String> roles = new ArrayList<String>();
```

Comment:
We avoid using mutable list as much as possible. Please use ImmutableList.Builder instead.

Reply:
@ebyhr would using ImmutableList.Builder look like this?
```
        ImmutableList.Builder<String> rolesBuilder = new ImmutableList.Builder<>();
        if (securityContext.isUserInRole("ADMIN")) {
            rolesBuilder.add("ADMIN");
        }
        if (securityContext.isUserInRole("USER")) {
            rolesBuilder.add("USER");
        }
        if (securityContext.isUserInRole("API")) {
            rolesBuilder.add("API");
        }
        List<String> roles = rolesBuilder.build();
```